src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -369,3 +369,29 @@
                                      :receive-token-symbol receive-token-symbol
                                      :receive-amount       receive-amount})
                         (i18n/label :t/swap-failed))}]]]}))))
+
+(rf/reg-event-fx :wallet.swap/flip-assets
+ (fn [{:keys [db]}]
+   (let [{:keys [asset-to-pay asset-to-receive
+                 swap-proposal amount]} (get-in db [:wallet :ui :swap])
+         receive-token-decimals         (:decimals asset-to-receive)
+         amount-out                     (when swap-proposal (:amount-out swap-proposal))
+         receive-amount                 (when amount-out
+                                          (number/remove-trailing-zeroes
+                                           (.toFixed (number/convert-to-whole-number
+                                                      (native-module/hex-to-number
+                                                       (hex/normalize-hex
+                                                        amount-out))
+                                                      receive-token-decimals)
+                                                     receive-token-decimals)))]
+     {:db (update-in db
```

Comment:
Leaning on the side of personal preference :) but is it worth the extra indentation and the usage of an anonymous function just to not repeat `[:wallet :ui :swap]` once?

```clojure
{:db (-> db
         (update-in [:wallet :ui :swap]
                    assoc
                    :asset-to-pay     asset-to-receive
                    :asset-to-receive asset-to-pay
                    :amount           (or receive-amount amount))
         (update-in [:wallet :ui :swap]
                    dissoc
                    :swap-proposal
                    :error-response
                    :loading-swap-proposal?
                    :last-request-uuid))}
```